### DISCUSSION TITLE
Cannot Install Ubuntu, reboot returns to install screen

### DISCUSSION THREAD

**Discussion by charlescurt** [utmapp/UTM#3716] [UTM Help (macOS)]
I am using a mac with M1 and none of the install steps work for me. UTM will open Ubuntu and go through install but on reboot, it brings me right back to the beginning. It will also often crash mid-install steps. 

I have also noticed that my UTM looks different than the tutorials, is there an older version of UTM I can try to use? I am using UTM 3.0.4

<img width="208" alt="Screen Shot 2022-03-03 at 11 19 31 AM" src="https://user-images.githubusercontent.com/98784768/156636906-918293e2-4eba-493a-b36e-48d241ec6a84.png">
<img width="796" alt="Screen Shot 2022-03-03 at 11 17 45 AM" src="https://user-images.githubusercontent.com/98784768/156636919-920df6c0-b965-42b7-9ff9-8ea75f9aa0d9.png">
<img width="811" alt="Screen Shot 2022-03-03 at 11 17 56 AM" src="https://user-images.githubusercontent.com/98784768/156636931-49e9c790-28c9-444d-b799-04f1124ebcd2.png">

These are the tutorials and issues I looked through trying to solve this:
https://www.youtube.com/watch?v=hnwK-nkXolc
https://poolideas.cn/2021/12/21/how_to_install_linux_on_an_m1_mac_with_apple_silicon/
https://linuxhint.com/run-linux-m1-based-mac/

https://github.com/utmapp/UTM/issues/3225
https://github.com/utmapp/UTM/discussions/2597
https://github.com/utmapp/UTM/issues/2333

**Comment by emreunsal33** [comment]
I have the same problem. After the installation is done and click to rebot. Just wait in the boot screen and do noting. Black screen only

Macbook air M1 16GB ram 512 SSD
UTM 3.0.4
Ubuntu 20.04 LTS arm64 server

**Comment by radiorpl** [comment]
same problem, after install reboot does not work. when i shut down and restart the machine, it just goes back to install menu.
Macbook air M1 16GB ram 512 SSD
UTM 3.0.4
Ubuntu 20.04 LTS arm64 desktop

**Comment by radiorpl** [comment]
I had the same issue when installing kali linux, then i just ejected image from cd/dvd drive before next boot and it went to install. i haven't tried with ubuntu, but maybe try that.

**Reply by lcsmas** [reply]
Ejecting the installation iso worked for me with Ubuntu 24

**Reply by ITpandaffm** [reply]
It works!  just eject( clear CD/DVD)  THANKS

**Reply by pkasante** [reply]
It really works. After looping through multiple installations, this was the simple solution I needed. Thanks!

**Comment by niklr** [comment]
Same issue with:

Macbook Pro M1 Max (Apple Silicon)
UTM 3.1.4
Ubuntu 20.04 LTS arm64 server

**Reply by niklr** [reply]
Got it working by disabling the network before installing. Seems to be related to https://github.com/utmapp/UTM/issues/2682#issuecomment-1064827650

**Comment by kh398927** [comment]
Similar issue with "m2" MacBook Air (M2, 2022)

I am trying to install Ubuntu in UTM on MacBook Air m2. 
UTM: 3.2.4 (58)
Ubuntu: ubuntu-22.04.1-live-server-arm64.iso

Installation looks complete but it won't complete:
- "Reboot Now" only shows blinking cursor
- Only option is to close the window, to have "Confirmation - Closing this window will kill the VM."
- Returns to installation screen (GNU GRUB version 2.06)

Installation has complete with "m1" MacBook Air (M1, 2020)

Any comment or help would be appreciated.

![MacBook_Air_m2_specs](https://user-images.githubusercontent.com/38565789/195962227-56b77032-89d8-42ef-b739-83a1e60b6cc3.jpg)
![Screenshot 2022-10-15 at 9 58 11](https://user-images.githubusercontent.com/38565789/195962396-f3a6b9ac-dd8f-4fbc-9ac8-6b22394e721a.jpg)
![Screenshot 2022-10-15 at 9 59 48](https://user-images.githubusercontent.com/38565789/195962420-5b22dc1a-e123-47de-9743-8a876f0fc137.jpg)
![Screenshot 2022-10-15 at 10 01 17](https://user-images.githubusercontent.com/38565789/195962478-873ac75a-4277-4124-a13f-350617e85da7.jpg)
![Screenshot 2022-10-15 at 10 00 57](https://user-images.githubusercontent.com/38565789/195962537-db71b090-f814-4da9-8093-d7de7bfb8e29.jpg)

**Comment by gkhull** [comment]
This non server iso worked for me. Give it a try. 
https://cdimage.ubuntu.com/jammy/daily-live/current/jammy-desktop-arm64.iso

**Reply by kh398927** [reply]
Thanks! I will do it now.

**Reply by kh398927** [reply]
I tried, but was stuck when asked to fill in Username and Password.
(1)What is Username supposed to be?  I put some new name for it instead of the username to login to MacBook.  Is it OK?
(2)What is Password supposed to be?  I thought I was defining it now. Is it OK?
(3) Putting the password yields an error message "Sorry, password authentication didn't work. Please try again." endlessly.

Does anything ring a bell , please?
![Screenshot 2022-10-15 at 22 46 27](https://user-images.githubusercontent.com/38565789/195990479-9a02fcb3-615e-48fe-b424-32cf5af326a1.jpg)
![Screenshot 2022-10-15 at 22 44 39](https://user-images.githubusercontent.com/38565789/195990485-e19943cc-7003-456a-b33b-46c1eae51d11.jpg)
![Screenshot 2022-10-15 at 22 45 23](https://user-images.githubusercontent.com/38565789/195990492-7259ac87-ef31-48da-8768-df63df74f260.jpg)

**Reply by gkhull** [reply]
Its literally just
ubuntu
ubuntu

**Reply by kh398927** [reply]
With "ubuntu" as Username, I could step forward to installation, and Installation looked to have completed with a message "Installation Complete".  
Nevertheless, it does not seem a real end.  There were two options to choose: "Continue Testing" and "Restart Now".  Whichever option to choose, installation does not finish.  With "Continue Testing", nothing happens without black screen being turned up.  I had no option but to close the UTM window. With "Restart Now", I was led to the installation screen (GNU GRUB version 2.06) again.  

Any suggestion, please?
![Screenshot 2022-10-16 at 23 19 19](https://user-images.githubusercontent.com/38565789/196154473-66602d34-f0c4-4d35-bb45-f573bea96c56.jpg)
![Screenshot 2022-10-17 at 17 48 04](https://user-images.githubusercontent.com/38565789/196154500-3576421d-6c35-4136-9728-95f640a1e4c0.jpg)
![Screenshot 2022-10-17 at 18 18 25](https://user-images.githubusercontent.com/38565789/196154544-2cc0d069-f505-456c-93a1-b8a5939cb0d8.jpg)

**Comment by Andy312432** [comment]
Did you guys adjust the boot order after installing it?

**Reply by kh398927** [reply]
Thanks for the comment. But would you let me know how to do that in detail, please?

**Reply by Andy312432** [reply]
The device queue in the settings is your boot order

**Reply by Andy312432** [reply]
So you can just move that device to the much lower level

**Comment by mdluo** [comment]
Moving the **VirtIO Drive** on top of the **USB Drive** in the Edit > Drives worked for me:

<img width="976" alt="image" src="https://user-images.githubusercontent.com/4597409/198946008-9fab5bb6-5905-4ce4-ad61-3e144a155508.png">

**Reply by mfort-youngcheol** [reply]
@mdluo It worked for me as well, thanks for the information 👍

**Reply by 1eandro** [reply]
It worked for me 2

**Reply by voronindenis** [reply]
It worked for me too

**Reply by EMT-IOT** [reply]
super worked for mee tooo

**Comment by ravenxrz** [comment]
clear
<img width="1648" alt="image" src="https://user-images.githubusercontent.com/44746032/210140364-ece83bf3-31a8-40f3-9172-ed71173c51d9.png">
worked for me.

**Comment by aliridha1510** [comment]
Hey,

@mdluo @ravenxrz 

```
MacBook Pro
Chip Apple M1 Pro
Mac Os Venture 13.1
```

Ubuntu Virtual Machine no longer runs , before it ran normally without problems . now is only black screen.

I have moved VirtlDrive but not fixed,

does anyone have any idea help 

Thanks

1. 

<img width="751" alt="Screenshot 2023-01-17 at 11 48 41" src="https://user-images.githubusercontent.com/23260197/212879775-6b921f58-9726-43eb-85d7-f5eaab1ae46e.png">
2. 

![Screenshot 2023-01-17 at 11 13 29](https://user-images.githubusercontent.com/23260197/212871118-f9a20f68-f583-4ddd-9e11-7f7f7292ead5.png)
3. 

<img width="798" alt="Screenshot 2023-01-17 at 11 39 59" src="https://user-images.githubusercontent.com/23260197/212877794-d88e4059-0666-4d8a-bf1b-1f9007edc18b.png">

Settings :

<img width="807" alt="Screenshot 2023-01-17 at 11 36 59" src="https://user-images.githubusercontent.com/23260197/212877132-46bf253d-ecfb-4bc1-af86-ca3835d3e48b.png">

<img width="795" alt="Screenshot 2023-01-17 at 11 37 28" src="https://user-images.githubusercontent.com/23260197/212877251-6e36f7f9-8c0b-49cd-9076-4b77c1fbb823.png">

<img width="645" alt="Screenshot 2023-01-17 at 11 48 05" src="https://user-images.githubusercontent.com/23260197/212879631-f8639bda-ccec-4282-8072-00284d9809c3.png">

**Comment by sjarmund** [comment]
Hi, I have followed these steps. I manage to run linux ok. I have Ventura 13.1. Terminal is functioning great, but when I try to open firefox it starts lagging and don't open. And when I installed arduino IDE through the terminal, the program just shows black. Have this happened to anyone before?

<img width="1280" alt="Skjermbilde 2023-02-03 kl  10 05 23" src="https://user-images.githubusercontent.com/124348802/216557983-eca65f27-7d02-4c08-927f-693cc4ae26f9.png">

**Reply by BMilliet** [reply]
I had a similar problem with Fedora, could't run any GUI. What solved the issue for me was changing `settings > Display > Emulated Display Card > virtio-ramfb`

**Comment by an-dr32** [comment]
Maybe a little late for the party but, to solve the issue where you go back to the installer, just click on the Windows option at the top right and eject the installer after the installation is completed. An easy solution that was never explained.

**Reply by raweber42** [reply]
Very good point, works for me on M2 Macbook Pro

**Comment by becooq81** [comment]
> I am using a mac with M1 and none of the install steps work for me. UTM will open Ubuntu and go through install but on reboot, it brings me right back to the beginning. It will also often crash mid-install steps.
> 
> I have also noticed that my UTM looks different than the tutorials, is there an older version of UTM I can try to use? I am using UTM 3.0.4
> 
> <img alt="Screen Shot 2022-03-03 at 11 19 31 AM" width="208" src="https://user-images.githubusercontent.com/98784768/156636906-918293e2-4eba-493a-b36e-48d241ec6a84.png">
> 
> <img alt="Screen Shot 2022-03-03 at 11 17 45 AM" width="796" src="https://user-images.githubusercontent.com/98784768/156636919-920df6c0-b965-42b7-9ff9-8ea75f9aa0d9.png">
> 
> <img alt="Screen Shot 2022-03-03 at 11 17 56 AM" width="811" src="https://user-images.githubusercontent.com/98784768/156636931-49e9c790-28c9-444d-b799-04f1124ebcd2.png">
> 
> These are the tutorials and issues I looked through trying to solve this: https://www.youtube.com/watch?v=hnwK-nkXolc https://poolideas.cn/2021/12/21/how_to_install_linux_on_an_m1_mac_with_apple_silicon/ https://linuxhint.com/run-linux-m1-based-mac/
> 
> #3225 #2597 #2333

Click CD/DVD and empty it. This is the equivalent of ejecting the USB drive after installing Ubuntu.

**Comment by ja-colt** [comment]
After you install, shut the vm down. Click the box at the bottom where it says "CD/DVD" (see your second screenshot) and choose "Clear". Now boot the VM back up.

Basically you still have the installer "inserted" so it boots from that every time you start the VM. You just need to remove it.

**Reply by tylerarnold** [reply]
yes, this worked for me. let the VM be shut down or unmount the CD/DVD and boot.

**Comment by thomasriley0** [comment]
Everytime that I attempted to install debian bookwarm netinstall, it would simply just boot back into the install screen. The way that I fixed this was after completing the debian installation you need to go back to UTM and go to the box you are using to install debian. Then at the bottom, you click on CD/DVD then click clear. You may then simply proceed by rebooting and it will work this time.

**Comment by AnushangaWimalasena** [comment]
![image](https://github.com/user-attachments/assets/82ffca01-a6e0-478b-a56c-3cf0fe607507)

Eject CD/DVD and restart server. Worked for me.

Mac mini M2

**Comment by H3XploR** [comment]
eject the iso at the end.
